### PR TITLE
Update from voxpupuli modulesync_config

### PR DESCRIPTION
**Pull request by jyaworski**

### CODE REVIEW

**Comment by jyaworski** [comment]
Ping @dalisch @danifr @bastelfreak 

Jruby doesn't like the shell provider.

**Comment by rnelson0** [comment]
It's trying to fork /bin/sh and failing because jruby doesn't support that (
https://docs.puppet.com/puppet/latest/reference/type.html#exec-provider-shell).
Is there a reason that provider is used rather than the default posix
provider?

**Comment by jyaworski** [comment]
@rnelson0 weird that jruby doesn't support it. It seems like enough things use fork....

I can try to refactor it to use the posix provider.

**Comment by jyaworski** [comment]
Weird. I removed the shell provider, and the same error happens.

**Comment by igalic** [comment]
my suspicion would be that puppet turns a `command` with `;` into `shell` but i'll have to dig through the code to verify that.

File: manifests/config/file_keystore.pp [hunk 1]
```diff
@@ -72,15 +72,15 @@
   $meta_fqpath = "${file_keystorage_dir}/meta/keys/${path}"
 
   exec { "create ${path}_${name} key path":
-    provider => shell,
-    command  => "mkdir -m 775 -p ${key_fqpath}; chown -R ${user}:${group} ${key_fqpath}",
-    creates  => $key_fqpath,
+    path    => ['/usr/bin', '/usr/sbin', '/bin'],
```
Comment (igalic) [inline manifests/config/file_keystore.pp]: add `/usr/local/bin` here, please

File: manifests/config/file_keystore.pp [hunk 2]
```diff
@@ -72,15 +72,15 @@
...
+    command => "mkdir -m 775 -p ${key_fqpath}; chown -R ${user}:${group} ${key_fqpath}",
```
Comment (igalic) [inline manifests/config/file_keystore.pp]: if `chown`'s `-R` here is a red-herring, then please use `install -m 755 -o ${user} -g ${group} -d ${key_fqpath}`

Comment (jyaworski) [inline manifests/config/file_keystore.pp]: 👍 I'll change that.

Comment (jyaworski) [inline manifests/config/file_keystore.pp]: @igalic how's it look?

**Comment by jyaworski** [comment]
I did a pretty big refactor there, and it appears that the fork problem persists. Does it not like 'with_content'?

**Comment by jyaworski** [comment]
Now I'm getting jruby errors with rspec itself? autoload problems.

**Comment by bastelfreak** [comment]
jruby errors? travis looks fine, just rubocop is a bit pissed.

**Pull request merged**